### PR TITLE
@W-16549492 Conversation History Text and Attachment API schema and readme update

### PR DESCRIPTION
**Pull request by subhi1998**
Updated public repo with the ConversationHistory & AttachmentHistory API schema

WI - https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001zUxXXYA0/view

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+        },
+      "joinedTime":<Timestamp in Unix Epoch Milliseconds>
+    }
+  ],
+  "conversationEntries": [
+    {
+        "clientTimestamp": <Timestamp in Unix Epoch Milliseconds>,
+        "sender": {
+            "subject": "<unique identifier>",
+            "role": "ParticipantRole",
```
Comment (ssharma-salesforce) [inline README.md:434]: it should be `"role": "<ParticipantRole>"`

File: README.md [hunk 2]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+            "appType": "AppType"
```
Comment (ssharma-salesforce) [inline README.md:435]: this should be `"appType": "<AppType>"`

File: README.md [hunk 3]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+#### AttachmentsHistory API
+
+Example request for sending attachment History to MCS, be sure to replace placeholder values `<..>`
+
+```bash
+curl --location 'http://localhost:8085/api/v1/attachmentHistory' \
+--header 'OrgId: <orgId>>' \
+--header 'AuthorizationContext: <AuthorizationContext>' \
+--header 'RequestId: <RequestId>' \
+--header 'AuthorizationContextType: CONVERSATIONCHANNELDEFINITION' \
```
Comment (ssharma-salesforce) [inline README.md:463]: same as above

File: README.md [hunk 4]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+curl -v \
+-H "Authorization: Bearer <AccessToken>" \
+-H "content-type: application/json" \
+-H "OrgId: <OrgId>" \
+-H "AuthorizationContext: <AuthorizationContext>" \
+-H "AuthorizationContextType: CONVERSATIONCHANNELDEFINITION" \
+-H "RequestId: <RequestId>" \
+-X POST -d '{
+  "channelAddressIdentifier": "<channelAddressIdentifier>",
+  "conversationParticipants": [
```
Comment (ssharma-salesforce) [inline README.md:418]: why do we have a different variants of example. Other APIs are added with JSON encoded format but for this API, we have added JSON payload. Can you please ensure it's consistent to other APIs?

File: README.md [hunk 5]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
     }
   }
 }
+```  
+
+#### ConversationHistory API
+
+The API supports validation to ensure that partners don’t import conversations older than 24 hrs.
+
+Example request for sending conversation History to MCS, be sure to replace placeholder values `<..>`
```
[the remark this answers is not in the document]
Comment (subhi1998) [reply README.md]: updated.

File: README.md [hunk 6]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+
+```bash
+curl -v \
+-H "Authorization: Bearer <AccessToken>" \
+-H "content-type: application/json" \
+-H "OrgId: <OrgId>" \
+-H "AuthorizationContext: <AuthorizationContext>" \
+-H "AuthorizationContextType: CONVERSATIONCHANNELDEFINITION" \
```
[the remark this answers is not in the document]
Comment (subhi1998) [reply README.md]: updated.

File: README.md [hunk 7]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+-H "RequestId: <RequestId>" \
+-X POST -d '{
+  "channelAddressIdentifier": "<channelAddressIdentifier>",
+  "conversationParticipants": [
+    {
+      "displayName": "<displayName>",
+      "participant": {
+            "subject": "<participantSubject>",
```
[the remark this answers is not in the document]
Comment (subhi1998) [reply README.md]: updated.

File: README.md [hunk 8]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+            "role": "<participantRole>",
```
[the remark this answers is not in the document]
Comment (subhi1998) [reply README.md]: updated,

File: README.md [hunk 9]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+            "appType": "<participantAppType>"
```
[the remark this answers is not in the document]
Comment (subhi1998) [reply README.md]: updated.

File: README.md [hunk 10]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+        },
+      "joinedTime":<Timestamp in Unix Epoch>
+    }
+  ],
+  "conversationEntries": [
+    {
+        "clientTimestamp": <Timestamp in Unix Epoch Milliseconds>,
+        "sender": {
+            "subject": "<senderSubject>",
```
[the remark this answers is not in the document]
Comment (subhi1998) [reply README.md]: updated.

File: README.md [hunk 11]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+            "role": "<senderRole>",
+            "appType": "<senderAppType>"
+        },
+        "entryPayload": {
+            "entryType": "<entryType>",
+            "id": "<payloadId>",
+            "abstractMessage": {
+                "messageType": "<messageType>",
```
[the remark this answers is not in the document]
Comment (subhi1998) [reply README.md]: updated.

File: README.md [hunk 12]
```diff
@@ -397,6 +397,102 @@ Here `"subject":"005xx0000012345"` is the salesforce id of sender.
...
+                "id": "<messageId>",
+                "staticContent": {
+                    "formatType": "<formatType>",
```
[the remark this answers is not in the document]
Comment (subhi1998) [reply README.md]: updated.

File: README.md (hunk 1 above)
Comment (subhi1998) [reply README.md]: updated.

File: README.md (hunk 2 above)
Comment (subhi1998) [reply README.md]: updated.

File: README.md (hunk 3 above)
Comment (subhi1998) [reply README.md]: updated,.

File: README.md (hunk 4 above)
Comment (subhi1998) [reply README.md]: updated.

File: inbound-interaction-api.yaml [hunk 13]
```diff
@@ -1400,6 +1400,89 @@ components:
...
+        id:
+          type: string
+          example: 57904eb6-5352-4c5e-adf6-5f100572cf54
+        attachmentIndex:
+          type: integer
+        contentLength:
+          type: integer
+          description: "Indicates the size of the file data in bytes"
+        text:
+          type: "string"
```
Comment (ssharma-salesforce) [inline inbound-interaction-api.yaml:1479]: can we remove the double quotes?

Comment (subhi1998) [reply inbound-interaction-api.yaml]: updated.

**Pull request merged**